### DISCUSSION TITLE
All contents inside the bbox on the PDF Page. Positioning to another pdf

### DISCUSSION THREAD

**Discussion by hmertakyatan** [pymupdf/PyMuPDF#3746] [Looking for help]
I dimensioned the pages with bbox, keeping the existing dimensions with all contents. I would like to position them in another pdf. I have read the documentation but I have not found anything that can help. The input and expected outputs are attached. 
[input.pdf](https://github.com/user-attachments/files/16446080/input.pdf)

[desired_output.pdf](https://github.com/user-attachments/files/16446087/desired_output.pdf)

**Comment by hmertakyatan** [comment]
And my raw codes. I Forgot to add
def paginate_labels_same_dimension(input_pdf_path, output_pdf_path, output_page_width_mm, output_page_height_mm):
    reader = PdfReader(input_pdf_path)
    writer = PdfWriter()
    outer_bbox = pageinfo.get_outer_bbox(input_pdf_path, 0)
    bbox_width = converter.pt_to_mm(outer_bbox.width)
    bbox_height = converter.pt_to_mm(outer_bbox.height)
    print(f"Page  Bounding Box Width (mm): {bbox_width}, Bounding Box Height (mm): {bbox_height}")


    output_page_width_pts = converter.mm_to_pt(output_page_width_mm)
    output_page_height_pts = converter.mm_to_pt(output_page_height_mm)

    rows = int(output_page_height_mm // bbox_height)
    columns = int(output_page_width_mm // bbox_width)
    labels_per_page = rows * columns
    num_labels = len(reader.pages)
    num_pages = (num_labels + labels_per_page - 1)
    num_pages = math.ceil(num_labels / labels_per_page)
    label_index = 0
    i = 0
    for _ in range(num_pages):
        new_pdf_page = PageObject.create_blank_page(width=output_page_width_pts, height=output_page_height_pts)
        print("Blank page created.")
        
        horizontal_margin = (output_page_width_pts - (columns * outer_bbox.width)) / 2
        vertical_margin = (output_page_height_pts - (rows * outer_bbox.height)) / 2
        
        for row in range(rows):
            for column in range(columns):
                if label_index < num_labels:
                    label_page = reader.pages[label_index]

                    
                    label_index += 1
            
        new_pdf_page.compress_content_streams()
        writer.add_page(new_pdf_page)
        i += 1
        print(i, ".page done.")

    with open(output_pdf_path, "wb") as output_pdf:
        writer.write(output_pdf)

**Comment by JorjMcKie** [comment]
This is a variation of the problem solved in this example script: https://github.com/pymupdf/PyMuPDF-Utilities/blob/master/examples/combine-pages/combine.py.
There, every 2 input pages are combined in one output page.
Your case seems to be combining 55 input pages into 1.
